File: Practice/AOJ/DSL/DSL3_C/dsl3_c.py
# しゃくとりはO(N)なので数え上げられそう

n, q = map(int, input().split())
arr = list(map(int, input().split()))

for x in list(map(int, input().split())):
    ans = 0
    r = 0
    tmp = 0
    for l in range(n):
        while r < n and tmp <= x:
            tmp += arr[r]
            r += 1
            ans += 1
            if tmp == x:
                break
        print("l:r = {}:{} -> {} temp : {}".format(l, r, arr[l:r], tmp))
        if r == l:
            r += 1
        else:
            tmp -= arr[l]
    print(ans)

File: Practice/AOJ/DSL/DSL3_C/dsl3_c.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

var sc = bufio.NewScanner(os.Stdin)
var out = bufio.NewWriter(os.Stdout)

func main() {
	buf := make([]byte, 1024*1024)
	sc.Buffer(buf, bufio.MaxScanTokenSize)
	sc.Split(bufio.ScanWords)
	defer out.Flush()
	N, Q := nextInt(), nextInt()
	A := nextInts(N)
	for _, X := range nextInts(Q) {
		ans, curSum, r := 0, 0, 0
		for l := 0; l < N; {
			for ; r < N && curSum+A[r] <= X; r++ {
				curSum += A[r]
			}
			if r-l > 0 {
				ans += r - l
				curSum -= A[l]
			}
			if r == l {
				r++
			}
			l++
		}
		fmt.Fprintln(out, ans)
	}
}

func next() string {
	sc.Scan()
	return sc.Text()
}

func nextInt() int {
	a, _ := strconv.Atoi(next())
	return a
}

func nextInts(n int) []int {
	ret := make([]int, n)
	for i := 0; i < n; i++ {
		ret[i] = nextInt()
	}
	return ret
}
